capstone-frontend/src/components/CreateList.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+import { Button, Form, Spinner } from 'react-bootstrap'
+
+class CreateList extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      loading: false
+    }
+  }
+
+  handleSubmit = async (event) => {
+
+    this.setState({ loading: true })
+
+    event.preventDefault();
+
+    const name = event.target[0].value
+
+    const newBooklist = {
+      "userID": window.localStorage.getItem("userID"),
+      "name": name,
+    }
+
+    // Store BookList in Firebase
+    await fetch("/api/booklist", {
+      method: "POST",
+      body: JSON.stringify(newBooklist)
+    });
+
+    const createdBookList = await fetch(`/api/booklist?userID=${window.localStorage.getItem("userID")}&name=${name}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    console.log(createdBookList)
+
+    document.location.href = `/listpage/${createdBookList[0].id}`;
```

Comment:
Doesn't @steven-solar 's POST return the id in the response of the POST request? Maybe you could get the json of that which would avoid your GET request 33-35.

Reply:
If you store the post fetch response, you can just call response.id to get the ID.

Reply:
Done.